iterators_task_2/student/include/impl.h

```diff
@@ -13,35 +12,31 @@ struct accumulator : public std::iterator<std::output_iterator_tag, void, void,
     /** @todo Iterator operations */
 
 protected:
-    Container* container;
+    Container &container;
 public:
-    explicit accumulator(Container& con): container(&con) {};
-
-    accumulator& operator=(Container& conLR) {
-        container->insert(container->end(), conLR.begin(), conLR.end());
-        return *this;
-    }
-
-    accumulator& operator=(Container&& conRR) {
-        container->insert(container->end(),
-                          std::make_move_iterator(conRR.begin()),
-                          std::make_move_iterator(conRR.end()));
+    explicit accumulator(Container& con): container(con) {};
+
+    // ??? If I would like to specify the same value_type for another container. Is it correct?
```

Comment:
I think it is not a good practice, because of implicitly converted types. This will reduce possible applications of our template